crates/payload/basic/src/lib.rs

```diff
@@ -587,6 +640,72 @@ fn build_payload<Pool, Client>(
 
         let block_number = initialized_block_env.number.to::<u64>();
 
+        // Deposit transactions are always inserted at the top of the block built by the payload.
+        // These transactions are pre-paid on L1 and have no signature.
+        #[cfg(feature = "optimism")]
+        {
```

Comment:
I think we should rip this entire block into it's own function so we can unit test it like crazy

Reply:
agreed

Reply:
Broken out into its own `PayloadBuilder` implementation now that the job generator is generic over the builder itself. There's some code duplication here that we can work on, but it may be best to keep it duplicated so that breaking it up even more in the future isn't as painful.